### PR TITLE
PubSub E2E tests

### PR DESCRIPTION
**Pull request by nachocano**
- Adding E2E tests for PubSub source

### CODE REVIEW

File: test/e2e/test_pubsub.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+
+	"cloud.google.com/go/pubsub"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```
Comment (vaikas-google) [inline test/e2e/test_pubsub.go]: nit: says to uncomment, but it's already uncommented?

**Comment by vaikas-google** [comment]
/lgtm
/approval
/hold
remove the hold if you'd rather fix the nit in a followup

File: test/e2e/test_pubsub.go (hunk 1 above)
Comment (nachocano) [reply test/e2e/test_pubsub.go]: yes, copy/paste from other places... 
let me change the comment in a follow up, I'm adding more changes... will update the comment then... thanks!

**Comment by nachocano** [comment]
/hold cancel

updating the comment in the follow up PR that is coming in few minutes

**Comment by nachocano** [comment]
> /lgtm
> /approval
> /hold
> remove the hold if you'd rather fix the nit in a followup

@vaikas-google there is a typo in approve? Can you approve again...

**Pull request merged**